### PR TITLE
Fix Emote Boards

### PR DESCRIPTION
**Pull request by jkriste**
This PR attempts to fix the following issues:
- Messages that we do not have permissions to see will throw a 403 Forbidden, even if we can see it being edited.
- Threads causing an assertion error.
- Images aren't embedding properly.
- Posts aren't being updated with the new emote count.
- Link to original message on post is hard to click on mobile.
- Leaderboard entries are ascending rather than descending.

### CODE REVIEW

File: ClemBot.Api/ClemBot.Api.Core/Features/EmoteBoardPosts/Bot/Details.cs [hunk 1]
```diff
@@ -90,22 +90,27 @@ public async Task<QueryResult<List<EmoteBoardPostDto>>> Handle(Query request, Ca
...
             {
-                var channelMessageIds = _context.EmoteBoardMessages
+                var channelMessageIds = await _context.EmoteBoardMessages
                     .Where(m => m.EmoteBoardPostId == post.Id)
                     .Select(m => new
                     {
                         m.ChannelId,
                         m.MessageId
                     })
-                    .ToDictionary(kvp => kvp.ChannelId, kvp => kvp.MessageId);
```
Comment (Jay-Madden) [inline ClemBot.Api/ClemBot.Api.Core/Features/EmoteBoardPosts/Bot/Details.cs:100]: wow, dont know how i missed this one

File: ClemBot.Bot/bot/services/emote_board_service.py [hunk 2]
```diff
@@ -101,10 +101,15 @@ async def on_message_edit(self, event: RawMessageUpdateEvent) -> None:
...
 
-        channel = guild.get_channel(event.channel_id)
+        channel = guild.get_channel_or_thread(event.channel_id)
         assert channel is not None and isinstance(channel, discord.abc.Messageable)
 
-        message = await channel.fetch_message(event.message_id)
+        # Attempt to fetch the message, ignore if we do not have permissions.
+        message: discord.Message
+        try:
+            message = await channel.fetch_message(event.message_id)
```
Comment (Jay-Madden) [inline ClemBot.Bot/bot/services/emote_board_service.py:110]: we can still link it right? just have the comment say "bot not permitted to read message" or something

**Review by Jay-Madden — CHANGES_REQUESTED** [review]

File: ClemBot.Bot/bot/services/emote_board_service.py (hunk 2 above)
Comment (jkriste) [reply ClemBot.Bot/bot/services/emote_board_service.py:110]: Not that I'm aware of. It's honestly kind of a weird edge case... We can see the message was edited, but we cannot read the message itself... If we can't read the message, we can't read the reactions, and we cannot update the embed if we cannot read the message content. Honestly not sure how this came about.

Comment (Jay-Madden) [reply ClemBot.Bot/bot/services/emote_board_service.py:110]: might not have granted the bot message content in oauth perms

Comment (Jay-Madden) [reply ClemBot.Bot/bot/services/emote_board_service.py:110]: probably need a log statement here so we know it happens

info level log

**Review by Jay-Madden — APPROVED** [review]
Let's Git er dunnnn

**Pull request merged**